lib/puppet/util/windows/principal.rb

```diff
@@ -112,6 +113,11 @@ def self.lookup_account_sid(system_name = nil, sid_bytes)
               FFI::MemoryPointer.new(:uint32, 1) do |name_use_enum_ptr|
 
                 sid_ptr.write_array_of_uchar(sid_bytes)
+
+                if Puppet::Util::Windows::SID.IsValidSid(sid_ptr) == FFI::WIN32_FALSE
+                  raise Puppet::Util::Windows::Error.new(_('Byte array for lookup_account_sid is invalid: %{sid_bytes}') % { sid_bytes: sid_bytes }, ERROR_INVALID_PARAMETER)
```

Comment:
Probably shouldn’t print the binary array. Or hex encode it?

Reply:
Similar problem in https://github.com/puppetlabs/puppet/pull/8202/files/54a14b291bb1f749961a5a161fba7efc33128556#diff-88ba5e4691d33bdb2fd674511129fa6fR126

and https://github.com/puppetlabs/puppet/pull/8202/files/54a14b291bb1f749961a5a161fba7efc33128556#diff-88ba5e4691d33bdb2fd674511129fa6fR133

Reply:
This actually works ok: 

```
irb(main):003:0> Puppet::Util::Windows::SID.octet_string_to_principal([2, 3, 4, 5, 6])
Puppet::Util::Windows::Error: Byte array for lookup_account_sid is invalid: [2, 3, 4, 5, 6]:  The parameter is incorrect.
```